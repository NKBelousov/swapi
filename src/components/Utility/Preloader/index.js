import React, { Component } from "react";
import PropTypes from "prop-types";

import styled, { keyframes } from "styled-components";

const Wrapper = styled.div`
  display: block;
  margin: 0 auto;
  text-align: center;
`;

const Text = styled.span`
  color: ${props => props.theme.primary};
  font-size: 2rem;
  padding: 1rem;
`;

const rotate = keyframes`
  from {
    transform: rotate(0deg);
  } to {
    transform: rotate(360deg);
  }
`;

const Loader = styled.div`
  animation: ${rotate} 2s linear infinite;
  border: 4px solid ${props => props.theme.secondary};
  border-radius: 50%;
  border-top: 4px solid ${props => props.theme.primary};
  display: inline-block;
  height: 1rem;
  transition: transform 0.3s linear;
  width: 1rem;
`;

class Preloader extends Component {
  static get propTypes() {
    return {
      text: PropTypes.string,
    };
  }
  static get defaultProps() {
    return {
      text: "Loading...",
    };
  }
  render() {
    return (
      <Wrapper>
        <Loader />
        <Text>{this.props.text}</Text>
      </Wrapper>
    );
  }
}

export default Preloader;
