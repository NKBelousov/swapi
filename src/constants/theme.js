export const PRIMARY = "#db2c2c";
export const SECONDARY = "#fff";

export default Object.freeze({
  primary: PRIMARY,
  secondary: SECONDARY,
});
