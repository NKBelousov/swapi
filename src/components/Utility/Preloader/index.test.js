import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import Preloader from "./index.js";
import theme from "~/constants/theme";

test("Should render correctly", () => {
  const text = "Preloader";
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Preloader>{text}</Preloader>
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
