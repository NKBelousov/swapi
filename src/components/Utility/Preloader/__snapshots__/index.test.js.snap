// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  className="sc-bdVaJa jXvXbQ"
>
  <div
    className="sc-htpNat kQtlTe"
  />
  <span
    className="sc-bwzfXH geNCSP"
  >
    Preloader
  </span>
</div>
`;
